synapse/tests/test_common.py

```diff
@@ -57,3 +57,47 @@ def test_common_listdir(self):
 
             retn = tuple(listdir(dirn, glob='*.txt'))
             self.eq(retn, ((path,)))
+
+    def test_common_chunks(self):
+        s = '123456789'
+        parts = [chunk for chunk in chunks(s, 2)]
+        self.eq(parts, ['12', '34', '56', '78', '9'])
+
+        parts = [chunk for chunk in chunks(s, 100000)]
+        self.eq(parts, [s])
+
+        parts = [chunk for chunk in chunks(b'', 10000)]
+        self.eq(parts, [b''])
+
+        parts = [chunk for chunk in chunks([], 10000)]
+        self.eq(parts, [[]])
+
+        parts = [chunk for chunk in chunks('', 10000)]
+        self.eq(parts, [''])
+
+        parts = [chunk for chunk in chunks([1, 2, 3, 4, 5], 2)]
+        self.eq(parts, [[1, 2], [3, 4], [5]])
+
+        # set is unslicable
+        tpass = False
```

Comment:
Can you just use `self.raises(...` instead of these `tpass`es?

Reply:
Have to use assertRaises() directly as a context manager, since we can't call chunks() directly since it is a generator that needs to be consumed in order to have the exception thrown.  I replaced this with assertRaises.